battleShip.py

```diff
@@ -0,0 +1,280 @@
+                                break
+                            elif saveChoice.lower() == 'q':
+                                exit()
+                            else:
+                                print("\t\tInvalid choice. Try again :<")
+                                continue
+                elif choice == '2':
+                    p = input("\t\tEnter your name :) ")
+                    player1 = Player(p)
+                    player2 = Player("Computer")  # Create a computer player
+
+                    player1.createGrid(26)
+                    player2.createGrid(26)
+
+                    player1.placeShips(6, 7, 26)
+                    player2.placeShips(6, 7, 26)
+
+                    while True:
+                        print(f"\n\t\t---- {player1.name} Turn ----")
+                        print(f"\t\t{player1.name} Grid: \n")
+                        player1.displayGrid()
+                        player1.attack(player2)
+
+                        if not player2.ships:
+                            print(f"\t\t\tKudos {player1.name} You wins! :>")
+                            break
+
+                        print("\n\t\t---- Computer's Turn ----")  # Computer player's turn
+                        player2.attack(player1)
+
+                        if not player1.ships:
+                            print("\t\t\tComputer wins :>\n Better luck next time :<")
+                            break
+
+                        saveChoice = input("\n\t\tDo you want to save the game? (y/n) :> ")
+                        if saveChoice.lower() == 'y':
+                            saveGame(player1,player2)
+                            break
+                elif choice == '3':
+                    break
+                else:
+                    print("\n\t\tInvalid choice. Try again :<")
+
+        elif choice == '2':
+            saved_game = loadGame()
+            if saved_game:
+                player1 = saved_game[0]
+                player2 = saved_game[1]
+
+                player2.createGrid(26)
+                player2.placeShips(6, 7, 26)
+
+                while True:
+                    print(f"\n\t\t---- {player1.name} Turn ----")
+                    print(f"\t\t{player1.name} Grid: \n")
+                    player1.displayGrid()
+                    player1.attack(player2)
+
+                    if not player2.ships:
+                        print(f"\n\t\tKudos {player1.name} You wins! :>)")
+                        break
+
+                    print(f"\n\t\t---- {player2.name} Turn ----")
+                    print(f"\t\t{player2.name} Grid: \n")
+                    player2.displayGrid()
+                    player2.attack(player1)
+
+                    if not player1.ships:
+                        print(f"\n\t\tKudos {player2.name} You wins! :>)")
+                        break
+
+                    saveChoice = input("\n\t\tDo you want to save the game? (y/n) :> ")
```

Comment:
this will ask user to quit after each step, that is not very user friendly, can you implement in some other way in which both, coordinates and custom commands are dealt in single line

i.e 

```
Enter choice (quit/save/attack <co-ordinates>): 
```

and instead of using numbers for both rows and columns can you think an efficient way of taking coordinate as single input 

i.e

```
Enter choice (quit/save/attack <co-ordinates>):  attack A24
Enter choice (quit/save/attack <co-ordinates>):  attack C26
```